### PR TITLE
sys-auth/sssd: Bump add v1.16.3 and v2.0.0

### PR DESCRIPTION
**Pull request by webknjaz**
This ebuild hasn't been upgraded for a while and I've spent lots of time googling about my problem. Turn out that `krb5_map_user` was broken and been fixed in v1.14. I assumed my version to be up-to-date earlier, but in fact it hasn't been bumped for a while.

P.S. Maintainers are welcome to push to my branch :)

### CODE REVIEW

File: sys-auth/sssd/sssd-1.16.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,208 @@
+# Copyright 1999-2017 Gentoo Foundation
```
Comment (Zlogene) [inline sys-auth/sssd/sssd-1.16.3.ebuild]: you do not need this version at all

File: sys-auth/sssd/sssd-2.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,208 @@
+# Copyright 1999-2017 Gentoo Foundation
```
Comment (Zlogene) [inline sys-auth/sssd/sssd-2.0.0.ebuild]: your calendar is off

File: sys-auth/sssd/sssd-2.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Comment (Zlogene) [inline sys-auth/sssd/sssd-2.0.0.ebuild]: 6

File: sys-auth/sssd/sssd-2.0.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6} )
```
Comment (Zlogene) [inline sys-auth/sssd/sssd-2.0.0.ebuild]: it supports 3.7 now

File: sys-auth/sssd/sssd-2.0.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+inherit eutils multilib pam linux-info autotools multilib-minimal python-r1 systemd toolchain-funcs
```
Comment (Zlogene) [inline sys-auth/sssd/sssd-2.0.0.ebuild]: eutils is not needed

File: sys-auth/sssd/sssd-2.0.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+	linux-info_pkg_setup
+}
+
+src_prepare() {
+	eautoreconf
+
+	multilib_copy_sources
+
+	# Maybe run it before eautoreconf?
+	epatch_user
```
Comment (Zlogene) [inline sys-auth/sssd/sssd-2.0.0.ebuild]: just drop it as I did for my local copy, I barely found users aoolying their own patches

File: sys-auth/sssd/sssd-2.0.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+multilib_src_install() {
+	if multilib_is_native_abi; then
+		emake -j1 DESTDIR="${D}" "${_at_args[@]}" install
+	else
+		# easier than playing with automake...
+		dopammod .libs/pam_sss.so
+
+		into /
+		dolib .libs/libnss_sss.so*
```
Comment (Zlogene) [inline sys-auth/sssd/sssd-2.0.0.ebuild]: `dolib.so`

File: sys-auth/sssd/sssd-2.0.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+		if use locator; then
+			exeinto /usr/$(get_libdir)/krb5/plugins/libkrb5
+			doexe .libs/sssd_krb5_locator_plugin.so
+		fi
+	fi
+}
+
+multilib_src_install_all() {
+	einstalldocs
+	prune_libtool_files --all
```
Comment (Zlogene) [inline sys-auth/sssd/sssd-2.0.0.ebuild]: this one is obsolete, please use `find` on `${D}` for la files and delete it

File: sys-auth/sssd/sssd-1.16.3.ebuild (hunk 1 above)
Comment (webknjaz) [reply sys-auth/sssd/sssd-1.16.3.ebuild]: Why not? Last version in previous major + current major

File: sys-auth/sssd/sssd-2.0.0.ebuild (hunk 6 above)
Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: Well, I know users doing so. What will this affect?

File: sys-auth/sssd/sssd-2.0.0.ebuild (hunk 7 above)
Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: More detail?

File: sys-auth/sssd/sssd-2.0.0.ebuild (hunk 8 above)
Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: I've just copy-pasted previous ebuild version. I don't know how to do this. Help me on this, please.

File: sys-auth/sssd/sssd-2.0.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+		--localstatedir="${EPREFIX}"/var
+		--enable-nsslibdir="${EPREFIX}"/$(get_libdir)
+		--with-plugin-path="${EPREFIX}"/usr/$(get_libdir)/sssd
+		--enable-pammoddir="${EPREFIX}"/$(getpam_mod_dir)
+		--with-ldb-lib-dir="${EPREFIX}"/usr/$(get_libdir)/samba/ldb
+		--without-nscd
+		--with-unicode-lib="glib2"
+		--disable-rpath
+		--disable-silent-rules
+		--sbindir=/usr/sbin
```
Comment (Dessa) [inline sys-auth/sssd/sssd-2.0.0.ebuild]: i'd like to see either:
a) a useflag for --enable-local-provider and enable-local-domain
b) both hardcoded

local domain is going away in a later major version, but maybe there are still people that use it?

Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: I don't really care, I just needed to fix an issue on my machine and contribute it back to upstream :)
I'll do as you want if you explain what to do exactly.

Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: @Dessa could you please send a PR or a snippet with that?

File: sys-auth/sssd/sssd-2.0.0.ebuild (hunk 8 above)
Comment (Zlogene) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: sorry, but a sssd is not my prime gentoo thingy to maintain in spite I am using it. You can always consult devmanual.gentoo.org to get details

Comment (Zlogene) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: @webknjaz 

you can make python compat shorter like `PYTHON_COMPAT=( python2_7 python3_{4,5,6,7} )` not to list all the impls from scratch.

Also could you please squash your commits into the single one with `git rebase -i origin/master` and `git push -f` then?

Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: Why? Don't you use "Squash and merge" mode when merging PR? Modern tools allow avoid having to manually squash things for merge. Anything blocking you from doing so? How do you do merges?

Comment (a17r) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: We rebase your PR on top of master, where your PR may consist of one or multiple atomic changes/commits (each of them complying to our git workflow: https://wiki.gentoo.org/wiki/Gentoo_git_workflow#Commit_policy). The preparation for this is done by the PR author, as doing this for each and every PR we review would not scale.

Comment (Zlogene) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: @webknjaz also your `PYTHON_COMPAT` is still wrong it is not the range, but the list.

File: sys-auth/sssd/sssd-2.0.0.ebuild (hunk 3 above)
Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: Why?

File: sys-auth/sssd/sssd-2.0.0.ebuild (hunk 8 above)
Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: @a17r so it's fine to have a few commits as long as they are atomic? I don't really like shuffle non-related things in one commit.

Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: @Zlogene why? {n1..nN} is a bash syntax, which unpacks to list, just like you want

Comment (Zlogene) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: We do not use such a disgusting bashizm, we do it explicitly

File: sys-auth/sssd/sssd-2.0.0.ebuild (hunk 3 above)
Comment (a17r) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: You're commenting on an outdated comment.

File: sys-auth/sssd/sssd-2.0.0.ebuild (hunk 8 above)
Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: But you just asked me to use curly brackets syntax yourself!

Comment (Zlogene) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: But do you see the difference between what you implemented and what i wrote a few comments above?

Comment (webknjaz) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: Yes, but I disagree with biased opinions and I've reacted to all constructive concerns. Nothing left to do on my side.

Comment (Zlogene) [reply sys-auth/sssd/sssd-2.0.0.ebuild]: @webknjaz a bit of russian language, I see it going  to fit better.

Если ты хочешь помочь, то читай документацию. Предпологать, что слепой копипаст пройдет это глупо. В итоге мы тратим время друг-друга на изменения которые можно сделать за пару минут, это фигня, а не сотрудничество получается :)

</russian>

**Pull request closed without merging**